### ISSUE TITLE
Features dissappear or error when merging on Postgresql layer

### ISSUE THREAD

**Issue by Virvis** [qgis/QGIS#51142]
### What is the bug or the crash?

I'm lazy to write all same attributes every time when adding new features.
Therefore I'll copy attributes from existing features or merge feature if needed.
Adding new polygon (or line or point) and select existing one to take attributes.
Then when merge features or merge selected feature attributes will also update features ID, which should be unique.

Then I will get error if I only merger attributes:
"
ERROR: 2 feature(s) not added.
      PostGIS-error when added.... ERROR:  duplicate key value violates unique constraint "layerexample_pkey"
    DETAIL:  Key (id)=(15) already exists.
"

And if merge selected features the data will dissappear when merging and when saving, the whole feature dissappear without any trace on Postgresql database. Just like nothing been ever there. Only marking is next id-value is bigger than earlier.

Project settings is marked "Evaluate default values on provider side". 

If not un-checked this "Evaluate default values on provider side" error still stays but now I can see id valued as:
nextval('"2023".layerexample_id_seq'::regclass)

### Steps to reproduce the issue

1. Add postgis layer to map canvas
2. Create feature, add information you want
3. Save information
4. Create new feature without added date
5. Select new and old feature
6. Merge Feature Attributes, 
7. Select old features row with attributes you want to copy for new feature
8. Select "Use selected features attributes"
9. Now you will have both features with same id.... 
10. Not possible to save.

1. If you merge now the features it may dissappear depending on selection of  "Evaluate default values on provider side". 
2. One option not to loose your inserted features is cut and paste features once from and to canvas, then they will have new id's and no error should not happen.

### Versions

QGISin versio | 3.22.13-Białowieża
and also 3.22.10.

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

https://github.com/qgis/QGIS/issues/49477#issue-1318419369

**Comment by roya0045** [comment]
#51122 ?

**Comment by DelazJ** [comment]
> 7. Select "Use selected features attributes"
>
> 8. Now you will have both features with same id....

Should you not "skip" modification of the `id` field if you care about its uniqueness?

**Comment by Virvis** [comment]
> Should you not "skip" modification of the `id` field if you care about its uniqueness?

It cannot be skipped. And when selecting merging type; ID -column will get null value ("") -which is not allowed, or all polygons will get same value, which is not allowed either.

**Comment by DelazJ** [comment]
> It cannot be skipped. And when selecting merging type; ID -column will get null value ("")

@Virvis not sure whether I understand. You are doing "Merge Feature Attributes" in order to share some attributes and avoid typing multiple times. Great. Do you state that if you select "Skip attribute" for the ID field, "output features" have that field emptied?

**Comment by Virvis** [comment]
Hmm. Think so. Maybe I am not sure how to "skip" only one attribute, when merging attributes. 

But found one way to work around this problem:
I was able to set trigger to fix this at our psql data server with code under:


CREATE OR REPLACE FUNCTION "2023".layerexample_id() RETURNS trigger AS
$$
BEGIN
 IF EXISTS (SELECT 1 FROM "2023".layerexample WHERE id is not null ) THEN
    NEW.id := nextval('"2023".layerexample_id_seq'::regclass);
 END IF;
 RETURN NEW;
  END;
$$
LANGUAGE plpgsql SECURITY DEFINER;


CREATE TRIGGER layerexample_insert_id_trigger
Before INSERT ON "2023".layerexample
  FOR EACH ROW EXECUTE PROCEDURE "2023".layerexample_id();

**Comment by Virvis** [comment]
Yes. Done sometimes that also, but doesn't help. Then attributes got zero values and therefore wasn't helping. Also not helping in point when I'm lazy and want only select row to attributes and press enter. Now tested that again and it looks like it could help if skipped that row. But cannot confirm when I solved already from server head this problem.
